### PR TITLE
Configuration logic for dependencies

### LINKED ISSUE #247: depends-on logic for new format
as discussed in #245

### PR DESCRIPTION
**Pull request by yyadavalli**
Resolve #247

### CODE REVIEW

**Comment by yyadavalli** [comment]
@fommil please verify my changes in the tests. I have updated the `ensime-server` test, I am not sure I have it entirely correct. 
I am not clear on whether the same module with different config should be present or not. 
For example, with my change `server in compile` depends on `core in Compile, core in it, core in test`.  Please clarify.

**Comment by fommil** [comment]
oh wow, totally out of the blue awesomeness :smile: it's bedtime here I'll have to review it in the morning. There is no "unread" button in github so if I forget, please ping me.

File: src/main/scala/EnsimePlugin.scala [hunk 1]
```diff
@@ -400,9 +401,16 @@ object EnsimePlugin extends AutoPlugin {
...
     ).toSet ++ (ensimeUnmanagedJavadocArchives in config in projectRef).gimme
 
-    // needs to take configurations into account
-    // https://github.com/ensime/ensime-sbt/issues/247
-    val deps = project.dependencies.map(_.project.project).map { n => EnsimeProjectId(n, "compile") }
+    val pattern = "([A-Za-z]+)->([A-Za-z]+)".r
+
+    val deps = project.dependencies.map { d =>
+      val name = d.project.project
+      val config = d.configuration.getOrElse("compile") match {
```
Comment (fommil) [inline src/main/scala/EnsimePlugin.scala]: perhaps cleaner as `.get.collect { case IvyConfig(from, to) => to }.getOrElse("compile")`

**Review by fommil — CHANGES_REQUESTED** [review]
the output looks good, so I think the logic is correct!

File: src/main/scala/EnsimePlugin.scala (hunk 1 above)
Comment (yyadavalli) [reply src/main/scala/EnsimePlugin.scala]: noob question, isn't it a bad idea to call `.get` on a `Option` directly
How about
```scala
val config = d.configuration.map {
  case ivyConfig(from, to) => to
  case _ => "compile"
}.getOrElse("compile")
```

Comment (fommil) [reply src/main/scala/EnsimePlugin.scala]: oh, I thought it was a `Map`. If it's an option already then just remove `.get`. It's convention to use uppercase pattern matches though.

Comment (fommil) [reply src/main/scala/EnsimePlugin.scala]: btw, `.collect` will turn the `case _` into a `None` and then the fallback will be used.

Comment (yyadavalli) [reply src/main/scala/EnsimePlugin.scala]: Ok thanks, got it. I will update it

**Comment by fommil** [comment]
you forgot to compile that :stuck_out_tongue: so the formatter got you

**Comment by yyadavalli** [comment]
Cleaned and updated

**Comment by fommil** [comment]
awesome, thank you so much this blocks the feature that I'm hoping our new sponsored student can work on!

**Comment by yyadavalli** [comment]
Welcome, what is he planning to work on?

**Pull request merged**